pkg/ethutil/ethutil.go

```diff
@@ -106,6 +108,12 @@ func getInputIndex(
 		if err != nil {
 			return 0, fmt.Errorf("failed to parse input added event: %v", err)
 		}
+
+		inputAddedJsonBytes, err := json.Marshal(inputAdded)
```

Comment:
Shouldn't it fail when `err != nil`?

Reply:
I had mixed feelings about this. I will do a different log , but not fail, as this deserialization is only used for debug logging

Reply:
Removed the Json Marshal. It was only useful for a very specific need.